rawnode.go

```diff
@@ -136,6 +136,29 @@ func (rn *RawNode) Ready() Ready {
 	return rd
 }
 
+// FlowControl tunes the volume and types of messages that GetMessages call can
+// return to the application.
+type FlowControl struct {
```

Comment:
Did you explore pushing `FlowControl` as an interface into `etcd/raft`, similar to library's relationship with the `Storage` interface? I would imagine that this would allow us to hide the interactions with the optional flow control without needing to change the external interface of the library.

We probably still would want a `DisableEagerAppends` option so that the library could be configured to only consult `FlowControl` on calls to `Ready()`/`HasReady()`. These functions would then loop over the peers and construct messages for each according to the `FlowControl` state. Pushing this into `etcd/raft` would also facilitate fast-paths for replica states (e.g. followers) that cannot send MsgApps and need not consult `FlowControl`.

Reply:
Both ways do extend the surface of the library, but interfacing `FlowControl` makes it indirect and implicit. I'm slightly in favour of avoiding this inversion of control, although I'm still exploring options.